php_phongo.c

```diff
@@ -864,10 +907,10 @@ void php_phongo_read_preference_to_zval(zval *retval, const mongoc_read_prefs_t
 	}
 
 	if (!bson_empty0(tags)) {
+		/* Use PHONGO_TYPEMAP_NATIVE_ARRAY for the root type since tags is an
+		 * array; however, inner documents and arrays can use the default. */
```

Comment:
Can we be explicit about the document_type? i.e., by setting it to something like:

```
state.map.document_type = PHONGO_TYPEMAP_DEFAULT;
```